.circleci/config.yml

```diff
@@ -125,6 +134,40 @@ jobs:
       - store_test_results:
           path: ~/junit/
           destination: testartifacts
+  build:
+    machine: true
+    environment:
+      _JAVA_OPTIONS: "-Xms1024m -Xmx2048m"
+      _SERVICE_OCCURENCE: 5
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - spring-cloud-kubernetes-{{ .Branch }}-{{ checksum "pom.xml" }}
+            - spring-cloud-kubernetes-{{ .Branch }}
+            - spring-cloud-kubernetes
+      - run:
+          name: dependencies
+          command: |
+            ./mvnw -s .settings.xml -U --fail-never dependency:go-offline -Dservice.occurence=${_SERVICE_OCCURENCE} || true
+      - run:
+          name: build
+          command: |
+            ./mvnw -s .settings.xml clean install -Dservice.occurence=${_SERVICE_OCCURENCE} -DskipTests
+      - run:
+          name: save docker images
+          command: |
+            TAG=$(./mvnw -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)
+            echo $TAG
+            mkdir docker-images
+            docker save -o docker-images/spring-cloud-kubernetes-configuration-watcher.tar docker.io/springcloud/spring-cloud-kubernetes-configuration-watcher:${TAG}
+      - persist_to_workspace:
```

Comment:
I understand what this steps does, but I don't understand why we need it :( are we saving some builds in this manner? thank you

Reply:
Its a bit of a time hack.  Basically we are building these images during the build stage, so to avoid building them again for the integration tests, I save them off and load them again in the test stage.

Reply:
nice! makes total sense